### PR TITLE
[Maulik] Implement update table on schema update

### LINKED ISSUE #7: Update bigquery table schema on proto change

### PR DESCRIPTION
**Pull request by mauliksoneji**
Fix #7

### CODE REVIEW

File: src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        String protoMapping = getProtoMappingFromRemoteURL();
+        updateBigQuerySchema();
+        protoMappingConfig.setProperty("PROTO_COLUMN_MAPPING", protoMapping);
+        setProtoParser(protoMappingConfig.getProtoColumnMapping());
+    }
+
+    private ColumnMapping getProtoMapping() throws ExternalCallException {
+        if (protoMappingConfig.getProtoColumnMappingURL() != null) {
+            String protoMapping = this.getProtoMappingFromRemoteURL();
+            protoMappingConfig.setProperty("PROTO_COLUMN_MAPPING", protoMapping);
```
Comment (devdinu) [inline src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java]: We have to handle the scenario where `getProtoMappingFromRemoteUrl` throwing Exception

File: src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        this.stencilClient = stencilClient;
+        this.setProtoParser(getProtoMapping());
+    }
+
+    @Override
+    public void onProtoUpdate() {
+        log.info("updating bq table as {} proto was updated", this.getProto());
+        try {
+            this.updateProtoParser();
+        } catch (ExternalCallException e) {
```
Comment (devdinu) [inline src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java]: unless there's a conflict, we could call the methods directly, we can remove `this`.

On side note, we should make a list of conventions to follow.

File: src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java (hunk 1 above)
Comment (mauliksoneji) [reply src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java]: @devdinu the `getProtoMappingFromRemoteUrl` function is called twice.
1. At startup when we need to fetch proto mapping from remote url
2. When proto changes and we need to refetch the latest proto mapping.

We are handling case 2 in `onProtoUpdate` function which will be called if proto is updated.
IMO, for the first case we should not handle the error as this is non recoverable.
In first case we should propogate the error and stop the application.

File: src/main/java/com/gojek/beast/models/UpdateBQTableRequest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package com.gojek.beast.models;
```
Comment (devdinu) [inline src/main/java/com/gojek/beast/models/UpdateBQTableRequest.java]: You could use lombok `@Data or @Getter and @Setter``

**Pull request merged**